futures-core/src/lib.rs

```diff
@@ -12,8 +12,6 @@
 
 #[cfg(feature = "alloc")]
 extern crate alloc;
-#[cfg(feature = "std")]
-extern crate std;
```

Comment:
This is necessary to generate the appropriate error if the user accidentally enables the std feature.

Otherwise, we may run into trouble with users who accidentally rely on the fact that currently we are not using std and get angry that their builds are corrupted during future updates.